test/unit/org/apache/cassandra/io/DiskSpaceMetricsTest.java

```diff
@@ -109,15 +112,13 @@ public void testFlushSize() throws Throwable
         for (int i = 0; i < 3; i++)
             insertN(KEYSPACE_PER_TEST, cfs, 1000, 55);
 
-        cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.UNIT_TESTS);
-        int totalSize = 0;
-        final Set<SSTableReader> liveSSTables = cfs.getLiveSSTables();
+        final List<SSTableReader> liveSSTables = cfs.getLiveSSTables().stream()
+                                                    .sorted(SSTableReader.idComparator)
+                                                    .collect(Collectors.toList());
+        MovingAverage expectedMetrics = ExpMovingAverage.decayBy1000();
         for (SSTableReader rdr : liveSSTables)
-        {
-            totalSize += rdr.onDiskLength();
-        }
-        final int avgSize = totalSize / liveSSTables.size();
-        assertEquals(avgSize, cfs.metric.flushSizeOnDisk.get(), 0.05 * avgSize);
+            expectedMetrics.update(rdr.onDiskLength());
+        assertThat(cfs.metric.flushSizeOnDisk.get()).isEqualTo(expectedMetrics.get());
```

Comment:
Nit: I would give is some small leeway, e.g. `isCloselTo(expectedMetrics.get(), withinPercentage(0.001))`.

Reply:
we could do that, though, I've run 500 iterations of this test without that delta and it passed each time